File: app/src/main/java/com/eftimoff/bakingapp/app/BakingApplication.java
package com.eftimoff.bakingapp.app;

import android.app.Application;

import com.eftimoff.bakingapp.app.injection.AppComponent;
import com.eftimoff.bakingapp.app.injection.AppModule;
import com.eftimoff.bakingapp.app.injection.DaggerAppComponent;

public class BakingApplication extends Application {

    private static AppComponent component;

    @Override
    public void onCreate() {
        super.onCreate();

        component = DaggerAppComponent.builder()
                .appModule(new AppModule(this))
                .build();
    }

    public static AppComponent component() {
        return component;
    }
}

File: app/src/main/java/com/eftimoff/bakingapp/BakingApplication.kt
package com.eftimoff.bakingapp

import android.app.Application
import android.content.Context
import com.eftimoff.bakingapp.app.injection.AppComponent
import com.eftimoff.bakingapp.app.injection.AppModule
import com.eftimoff.bakingapp.app.injection.DaggerAppComponent

class BakingApplication : Application() {

    companion object {
        lateinit var component: AppComponent

        fun get(context: Context): AppComponent {
            return DaggerAppComponent.builder()
                    .appModule(AppModule(context))
                    .build()
        }
    }

    override fun onCreate() {
        super.onCreate()
        component = get(this)
    }


}